srv/salt/_modules/cephimages.py

```diff
@@ -6,20 +6,30 @@
 from glob import glob
 from subprocess import Popen, PIPE
 
+def _is_installed(name):
+    try:
+        devnull = open(os.devnull)
+        Popen([name], stdout=devnull, stderr=devnull).communicate()
+    except OSError as e:
+        if e.errno == os.errno.ENOENT:
+            return False
+    return True
 
 def list():
     """
     Find all rbd images
     """
+
     images = {}
-    proc = Popen([ 'rados', 'lspools' ], stdout=PIPE, stderr=PIPE)
-    for line in proc.stdout:
-	pool = line.rstrip('\n')
-	cmd = [ '/usr/bin/rbd', '-p', pool, 'ls' ]
-	rbd_proc = Popen(cmd, stdout=PIPE, stderr=PIPE)
-	for rbd_line in rbd_proc.stdout:
-	    if pool not in images:
-		images[pool] = []
-	    images[pool].append(rbd_line.rstrip('\n'))
+    if _is_installed('rados'):
+        proc = Popen([ 'rados', 'lspools' ], stdout=PIPE, stderr=PIPE)
+        for line in proc.stdout:
+            pool = line.rstrip('\n')
+            cmd = [ '/usr/bin/rbd', '-p', pool, 'ls' ]
+            rbd_proc = Popen(cmd, stdout=PIPE, stderr=PIPE)
+            for rbd_line in rbd_proc.stdout:
+                if pool not in images:
+                    images[pool] = []
+                images[pool].append(rbd_line.rstrip('\n'))
```

Comment:
we should probably used the librados and librbd python bindings in these cases, ok for now

Reply:
true. want to open an issue to track this, @theanalyst

Reply:
ack, created https://github.com/SUSE/DeepSea/issues/665

Reply:
I think we can already check in this with the python bindings. Something like this:

```
images = {}
if not _is_installed('ceph-common'):
    return images
import rados
import rbd
cluster = rados.Rados(conffile='/etc/ceph/ceph.conf')
cluster.connect()
try:
    for pool in cluster.list_pools():
        ioctx = cluster.open_ioctx(pool)
        try:
            rbd_inst = rbd.RBD()
            images[pool] = rbd_inst.list(ioctx)
        finally:
            ioctx.close()
finally:
    cluster.shutdown()
return images
```

Reply:
actually you can do a try catch on import itself, ie. 
```py
try:
    import rados
except ImportError:
...
```